### ISSUE TITLE
Pod status updates take longer to propagate to the API than necessary

### ISSUE THREAD

**Issue by smarterclayton** [kubernetes/kubernetes#116617]
### What happened?

#107897 originally identified a number of inefficiencies in how pod status is reported back to the API:

1. Pods are queued in a channel and then processed one by one - this causes HoL blocking and writes every observed status change, but pod status is level driven (split into #116615 for inclusion into #115331)
2. Status is a GET then PATCH, but we now have server side apply and using it would reduce 2 RT to 1 RT and reduce QPS
3. We treat all status updates the same, when in reality certain transitions should be prioritized (unready -> ready, Running -> Succeeded|Failed)
4. Status is single threaded, and we could easily try multiple
5. It's hard to know when our status cache has been brought up to date (so that we can bypass a SSA attempt), we could use ResourceVersion tracking for that (apimachinery proposed allowing it to be parsed)

In general these contribute to #113606 and should be improved.

/sig node
/sig scalability

### What did you expect to happen?

Faster

### How can we reproduce it (as minimally and precisely as possible)?

See kubelet logs.  Several e2e tests stress this flow, but we need a test that shows the improvements.

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by sftim** [comment]
> We treat all status updates the same, when in reality certain transitions should be prioritized (unready -> ready, Running -> Succeeded|Failed)

One key actual state report for me is to see that the kubelet has at least seen the pod and, if you like, acknowledged its responsibility for further status updates. I think that's the update to set `status.startTime`.

**Comment by smarterclayton** [comment]
Right, that's another good one.  Please add others.  Note however that it may take 50ms for a round trip of a pod status, and some containers might start and go ready within that time.  Would be good to think through how we want to bias towards these.

**Comment by sftim** [comment]
With SSA, the kubelet can fire off multiple `PATCH` requests that - the kubelet believes - can arrive in an arbitrary order at the API server and produce the right outcome. I mean that's the dream, right?

**Comment by smarterclayton** [comment]
Right - Kubelet would simply need to guarantee that no two SSAs for the same pod are inflight at once.

**Comment by smarterclayton** [comment]
Re: two inflight SSAs for non-conflicting operations to the same pod

For our own sanity, we probably don't want to have to reason about that while debugging :)

**Comment by SergeyKanzhelev** [comment]
/reopen
/remove-lifecycle rotten

still can be improved
